tests/e2e/tests/pilot/mesh_config_verify_test.go

```diff
@@ -0,0 +1,295 @@
+	if err == nil {
+		aggregatedAppEPs = aggregateAppEPs(primaryAppEPs, remoteAppEPs)
+		// Verify that the mesh contains endpoints from both the primary and the remote clusters
+		err = verifyMCMeshConfig(primaryPodNames, remotePodNames, aggregatedAppEPs)
+	}
+
+	if err != nil {
+		tc.Kube.RemoteAppManager.UndeployApp(tmpApp)
+		return err
+	}
+
+	// Remove the v3 deployment to app "c" in the remote cluster and verify the mesh configuration
+	log.Infof("After deleting c-v3, verify that c-v3 is deleted from the mesh")
+	if err = undeployApp(remoteCluster, "c-v3", tmpApp); err != nil {
+		return err
+	}
+
+	// Verify that proxy config changes back to before adding c-v3
+	remotePodNames, remoteAppEPs, err = util.GetAppPodsInfo(tc.Kube.Namespace, tc.Kube.Clusters[remoteCluster], "app")
+	if err == nil {
+		aggregatedAppEPs = aggregateAppEPs(primaryAppEPs, remoteAppEPs)
+		if err = verifyMCMeshConfig(primaryPodNames, remotePodNames, aggregatedAppEPs); err != nil {
+			return err
+		}
+	}
+
+	return nil
+}
+
+func verifyMeshConfig() error {
+	// Collect the pod names and app's endpoints
+	podNames, appEPs, err := util.GetAppPodsInfo(tc.Kube.Namespace, tc.Kube.KubeConfig, "app")
+	if err != nil {
+		return err
+	}
+
+	if err = verifyPods(tc.Kube.Istioctl, podNames, appEPs); err != nil {
+		return err
+	}
+
+	// Add a v3 deployment to app "c" and verify the mesh configuration
+	// Service 'c' has already been created earlier. Don't create it anymore.
+	tmpApp, err := deployApp(primaryCluster, "c-v3", "c", 80, 8080, 90, 9090, 70, 7070, "v3", true, false, true, false)
+	if err != nil {
+		return err
+	}
+
+	if podNames, appEPs, err = util.GetAppPodsInfo(tc.Kube.Namespace, tc.Kube.KubeConfig, "app"); err == nil {
+		err = verifyPods(tc.Kube.Istioctl, podNames, appEPs)
+	}
+
+	if err != nil {
+		tc.Kube.AppManager.UndeployApp(tmpApp)
+		return err
+	}
+
+	// Remove the v3 deployment to app "c" in the remote cluster and verify the mesh configuration
+	log.Infof("After deleting c-v3, verify that c-v3 is deleted from the mesh")
+	if err = undeployApp(primaryCluster, "c-v3", tmpApp); err != nil {
+		return err
+	}
+
+	if podNames, appEPs, err = util.GetAppPodsInfo(tc.Kube.Namespace, tc.Kube.KubeConfig, "app"); err != nil {
+		return err
+	}
+
+	// Verify that proxy config changes back to before adding c-v3
+	if err = verifyPods(tc.Kube.Istioctl, podNames, appEPs); err != nil {
+		return err
+	}
+	return nil
+}
+
+func verifyEndpoints(actualEps []string, proxyEps []string) bool {
+	sort.Strings(actualEps)
+	sort.Strings(proxyEps)
+	if len(actualEps) != len(proxyEps) {
+		return false
+	}
+	for i, ip := range actualEps {
+		if ip != proxyEps[i] {
+			return false
+		}
+	}
+	return true
+}
+
+func verifyPod(istioctl *framework.Istioctl, podName string, appEPs map[string][]string) error {
+	// Only verify app Pods that have a sidecar
+	apps := []string{"a-", "b-", "c-", "d-", "headless-"}
```

Comment:
It would be better to not hard code this here,  but derive this from the apps that are deployed during the test.   If the apps in the test change updating here might be lost.   This can be done in a subsequent PR though I wouldn't block the merge of this based on that.

Reply:
To avoid that problem, I think that I can define a variable that's initialized in getApps() and used in here.